airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -1834,8 +1846,16 @@ def execute(self, context: Dict):
         )
         job_id = job_object.reference.job_id
         self.log.info("Waiting for job %s to complete", job_id)
-        hook.wait_for_job(job_id=job_id, project_id=self.project_id, location=self.location)
-        self.log.info("Job completed successfully.")
+        if not self.asynchronous:
+            hook.wait_for_job(
+                job_id=job_id,
+                project_id=self.project_id,
+                location=self.location
+            )
+            self.log.info("Job completed successfully.")
+        else:
+            self.log.info("Job submitted successfully.")
+            return job_id
```

Comment:
```suggestion
        if not self.asynchronous:
            hook.wait_for_job(
                job_id=job_id,
                project_id=self.project_id,
                location=self.location
            )
        self.log.info("Job submitted successfully.")
        return job_id
```
`hook.wait_for_job` is blocking operation so this should do and it's simpler, WDYT?